azure-iot-device/tests/provisioning/pipeline/test_provisioning_pipeline.py

```diff
@@ -209,32 +194,30 @@ def test_pipeline_configuration(self, input_security_client, pipeline_configurat
     @pytest.mark.it(
         "Runs a Set SecurityClient Operation with the provided SecurityClient on the pipeline"
     )
-    def test_security_client_success(
-        self, mocker, params_security_clients, input_security_client, pipeline_configuration
-    ):
+    def test_security_client_success(self, mocker, input_security_client, pipeline_configuration):
         mocker.spy(pipeline_stages_base.PipelineRootStage, "run_op")
         pipeline = ProvisioningPipeline(input_security_client, pipeline_configuration)
 
         op = pipeline._pipeline.run_op.call_args[0][1]
         assert pipeline._pipeline.run_op.call_count == 1
-        assert isinstance(op, params_security_clients["set_args_op_class"])
+        if isinstance(input_security_client, X509SecurityClient):
+            assert isinstance(op, pipeline_ops_provisioning.SetX509SecurityClientOperation)
```

Comment:
In a normal circumstance I would suggest this is a problem - when you start seeing conditional statements in a test, that tends to be a sign something is wrong.

In this case, the reason is because of these two different security clients having different effects on program flow should probably indicate the test unit should be split to be one for each type of security client.

HOWEVER, because the next task I'm going to be taking on is to unify auths (and by extension, the security clients in provisioning), don't worry about it, fixing this here would be a lot of work to add something I'm just going to end up removing shortly.